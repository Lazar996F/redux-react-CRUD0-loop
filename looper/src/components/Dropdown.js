import React from "react";
import styled from "styled-components";

const Title = styled.div`
  position: absolute;
  top: 5px;
  font-size: 9px;
  color: gray;
  margin-left: 11px;
  pointer-events: none;
`;

const Arrow = styled.div`
  position: absolute;
  top: 20px;
  right: 15px;
  pointer-events: none;
  border-style: solid;
  border-width: 8px 5px 0px 5px;
  border-color: #7b7b7b transparent transparent transparent;
`;

const Select = styled.div`
  position: relative;
  display: inline-block;
  margin-bottom: 15px;
  width: 100%;

  select {
    font-family: "Arial";
    display: inline-block;
    width: 100%;
    cursor: pointer;
    padding: 9px 11px;
    outline: 0;
    padding-top: 16px;
    border: 1px solid #bebebe;
    border-radius: 5px;
    background: #fff;
    color: rgb(85, 84, 84);
    appearance: none;
    -webkit-appearance: none;
    -moz-appearance: none;
    font-size: 12px;
    font-weight: bold;
    padding-bottom: 4px;
  }
  select::-ms-expand {
    display: none;
  }
  select:hover,
  select:focus {
    color: #000000;
    background: linear-gradient(to bottom, #014584, #6a9dbc);
  }
  select:disabled {
    opacity: 0.1;
    pointer-events: none;
  }
  select:hover ~ .select_arrow,
  select:focus ~ .select_arrow {
    border-top-color: #000000;
  }
  select:disabled ~ .select_arrow {
    border-top-color: #cccccc;
  }
`;

function Dropdown(props) {
  const itemOptions = props.items.map((item, index) => (
    <option key={index} value={item.id}>
      {item.value}
    </option>
  ));
  return (
    <Select>
      <select className="test-select" onChange={props.handleSelected}>
        <option selected> </option>
        {itemOptions}
      </select>
      <Arrow />
      <Title>{props.subtitle}</Title>
    </Select>
  );
}

export default Dropdown;
