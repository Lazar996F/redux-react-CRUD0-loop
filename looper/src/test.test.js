import React from "react";
import { shallow } from "enzyme";
import Dropdown from "./components/Dropdown";
import toJson from "enzyme-to-json";

const mockitems = [
  { value: "val 1", id: "1" },
  { value: "val 2", id: "2" },
  { value: "val 3", id: "3" },
  { value: "val 4", id: "4" },
];

it("it should render correctly", () => {
  const wrapper = shallow(<Dropdown items={mockitems} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
const mockSelected = jest.fn();
it("it should call handleChange function when value has change", () => {
  const wrapper = shallow(
    <Dropdown handleSelected={mockSelected} items={mockitems} />
  );
  wrapper
    .find(".test-select")
    .simulate("change", { target: { value: "val 1" } });
  expect(mockSelected).toBeCalledWith({ target: { value: "val 1" } });
});

it("it should open dropdown when click", () => {
  const wrapper = shallow(<Dropdown items={mockitems} />);
  wrapper.find(".test-select").simulate("click");
});
