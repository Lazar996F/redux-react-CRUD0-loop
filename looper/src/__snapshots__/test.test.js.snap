// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render correctly  1`] = `
<styled.div>
  <select>
    <option
      key="0"
      value="1"
    >
      val 1
    </option>
    <option
      key="1"
      value="2"
    >
      val 1
    </option>
    <option
      key="2"
      value="3"
    >
      val 1
    </option>
    <option
      key="3"
      value="4"
    >
      val 1
    </option>
  </select>
  <styled.div />
  <styled.div />
</styled.div>
`;

exports[`it should render correctly 1`] = `
<styled.div>
  <select
    className="test-select"
  >
    <option
      key="0"
      value="1"
    >
      val 1
    </option>
    <option
      key="1"
      value="2"
    >
      val 2
    </option>
    <option
      key="2"
      value="3"
    >
      val 3
    </option>
    <option
      key="3"
      value="4"
    >
      val 4
    </option>
  </select>
  <styled.div />
  <styled.div />
</styled.div>
`;

exports[`it should render correctly when recording is not active 1`] = `
<styled.div>
  <select>
    <option
      key="0"
      value="1"
    >
      val 1
    </option>
    <option
      key="1"
      value="2"
    >
      val 1
    </option>
    <option
      key="2"
      value="3"
    >
      val 1
    </option>
    <option
      key="3"
      value="4"
    >
      val 1
    </option>
  </select>
  <styled.div />
  <styled.div />
</styled.div>
`;
